app/models/report.rb

```diff
@@ -14,4 +20,47 @@ def editable?(target_user)
   def created_on
     created_at.to_date
   end
+
+  def save_with_mentions
+    transaction do
+      save!
+      update_mentions!
+
+      true
+    end
+  rescue ActiveRecord::RecordInvalid
+    false
+  end
+
+  def update_with_mentions(params)
+    transaction do
+      update!(params)
+      update_mentions!
+
+      true
+    end
+  rescue ActiveRecord::RecordInvalid
+    false
+  end
+
+  private
+
+  def update_mentions!
+    mentioning_ids = content.scan(%r{http://localhost:3000/reports/(\d+)}).flatten.map(&:to_i).uniq
+
+    already_mentioning_ids = mentioning_reports.map(&:id)
```

Comment:
```suggestion
    already_mentioning_ids = mentioning_report_ids
```

で同じことができたはず

https://stackoverflow.com/a/34142633/1058763